wire-ios-sync-engine/Source/UserSession/Search/SearchDirectory.swift

```diff
@@ -25,24 +25,40 @@ import Foundation
     let transportSession: TransportSessionType
     var isTornDown = false
 
-    private var refetchIncompleteUserMetadata: RecurringAction?
-    private var refetchIncompleteConversationMetadata: RecurringAction?
+    private let refreshUsersMissingMetadataAction: RecurringAction
+    private let refreshConversationsMissingMetadataAction: RecurringAction
+    private let refreshTeamMetadataAction: RecurringAction
 
     deinit {
         assert(isTornDown, "`tearDown` must be called before SearchDirectory is deinitialized")
     }
 
     public convenience init(userSession: ZMUserSession) {
-        self.init(searchContext: userSession.searchManagedObjectContext, contextProvider: userSession, transportSession: userSession.transportSession)
-
-        self.refetchIncompleteUserMetadata = userSession.refreshUsersMissingMetadata()
-        self.refetchIncompleteConversationMetadata = userSession.refreshConversationsMissingMetadata()
+        self.init(
+            searchContext: userSession.searchManagedObjectContext,
+            contextProvider: userSession,
+            transportSession: userSession.transportSession,
+            refreshUsersMissingMetadataAction: userSession.refreshUsersMissingMetadataAction,
+            refreshConversationsMissingMetadataAction: userSession.refreshConversationsMissingMetadataAction,
+            refreshTeamMetadataAction: userSession.refreshTeamMetadataAction
```

Comment:
Do we really need to update team metadata when opening search? AFAIK it updates the team and logo which we don't display during the search.

Reply:
Removed.